src/fhirtypes/common.ts

```diff
@@ -205,6 +213,41 @@ export function replaceField(
   }
 }
 
+/**
+ * Adds Mixin rules onto a Profile, Extension, or Instance
+ * @param {Profile | Extension | Instance} fshDefinition - The definition to apply mixin rules on
+ * @param {FSHTank} tank - The FSHTank containing the fshDefinition
+ */
+export function applyMixinRules(
+  fshDefinition: Profile | Extension | Instance,
+  tank: FSHTank
+): void {
+  // Rules are added to beginning of rules array, so add the last mixin rules first
+  [...fshDefinition.mixins].reverse().forEach(mixinName => {
```

Comment:
Would it be better just to create a top-level `mixedInRules` array, traverse the mixins in order pushing rules onto that array, and then prepend the `mixedInRules` array to the `fshDefinition.rules`?  The advantages are only minor but I think they are:
* It's easier to read/understand than going through the mixins backward
* If multiple errors are logged, they'll be logged in the same order as the author mixed things in
* _Maybe_ more efficient to only prepend an array once instead of multiple times?  (Not sure, I'm making an assumption that prepending requires a whole new allocation whereas pushing doesn't -- but it may be a totally invalid assumption).

Anyway, this is only a light suggestion.  If you disagree at all then keep it as is.

Reply:
I think for bullet point 2 alone that's worth doing. I'll update.